dist/common/scripts/scylla_util.py

```diff
@@ -520,14 +519,36 @@ def get_tune_mode(nic):
         return 'sq_split'
 
 
-def create_perftune_conf(nic='eth0'):
-    if os.path.exists('/etc/scylla.d/perftune.yaml'):
-        return
-    mode = get_tune_mode(nic)
-    yaml = out('/opt/scylladb/scripts/perftune.py --tune net --nic "{nic}" --mode {mode} --dump-options-file'.format(nic=nic, mode=mode))
-    with open('/etc/scylla.d/perftune.yaml', 'w') as f:
-        f.write(yaml)
-
+def create_perftune_conf(cfg):
+    """
+    This function checks if a perftune configuration file should be created and
+    creates it if so is the case, returning a boolean accordingly. It returns False
+    if none of the perftune options are enabled in scylla_server file. If the perftune
+    configuration file already exists, none is created.
+    :return boolean indicating if perftune.py should be executed
+    """
+    params = ''
+    if get_set_nic_and_disks_config_value(cfg) == 'yes':
+        nic = cfg.get('IFNAME')
+        if not nic:
+            nic = 'eth0'
+        params += '--tune net --nic "{nic}"'.format(nic=nic)
+
+    if cfg.get('CLOCKSOURCE') == 'yes':
+        params += ' --tune-clock'
+
+    if len(params) > 0:
+        if os.path.exists('/etc/scylla.d/perftune.yaml'):
+            return True
```

Comment:
Hmmm... I'm not sure this is correct. What if the content of `perftune.yaml` was generated with one set of parameters and now you are given a different one? I think you should re-generate it regardless if it exists.

Reply:
We don't want to regenerate, Vlad.

What if the user has changed something manually? We don't regenerate today, as far as I understand, so I am not sure this behavior should change.

We can force regenerating the file by deleting it if anything changes. Regenerating unconditionally is dangerous. We wouldn't need a conf file if we were to always regenerate it.

Reply:
Yes, this is the current behavior. I didn't want to change it since it was not related to this patch.

Reply:
You are both right.